lib/global-admin/addon/components/form-global-roles/component.js

```diff
@@ -86,70 +79,75 @@ export default Component.extend({
   },
 
   save() {
-    this.set('errors', null);
-    let add         = [];
-    let remove      = [];
-
-    const user      = get(this, 'user');
-    const custom    = get(this, 'custom');
-    const baseRole  = get(this, 'allRoles').findBy('isBase', true);
-    const userRole  = get(this, 'allRoles').findBy('isUser', true);
-    const adminRole = get(this, 'allRoles').findBy('isAdmin', true);
-    const mode      = get(this, 'mode');
-
-    switch ( mode ) {
-    case ADMIN:
-
-      remove = custom.filterBy('existing').filter((x) => {
-        return get(x, 'role.isUser') || get(x, 'role.isBase') || get(x, 'role.isCustom');
-      }).map((x) => x.existing);
-
-      if ( !get(user, 'hasAdmin') ) {
-        add.push(this.make(adminRole));
-      }
-      break;
+    set(this, 'errors', null);
+    const add      = [];
+    const remove   = [];
+    const allRoles = [...this.baseRoles, ...this.additionalRoles];
 
-      // Admin intentionally doesn't remove any roles
+    add.pushObjects(allRoles.filterBy('active').filterBy('existing', false).map((r) => this.make(r.role)));
 
-    case USER:
-      remove = custom.filterBy('existing').filter((x) => {
-        return get(x, 'role.isAdmin') || get(x, 'role.isBase') || get(x, 'role.isCustom');
-      }).map((x) => x.existing);
+    remove.pushObjects(allRoles.filterBy('active', false).filterBy('existing').map((r) => r.existing));
 
-      if ( !get(user, 'hasUser') ) {
-        add.push(this.make(userRole))
-      }
-      break;
+    return PromiseAll(add.map((x) => x.save())).then(() => {
+      return PromiseAll(remove.map((x) => x.delete())).then(() => {
+        return true;
+      });
+    });
+  },
 
-    case CUSTOM:
-      add    = custom.filterBy('active', true ).filterBy('existing', null).map((x) => this.make(x.role));
-      remove = custom.filterBy('active', false).filterBy('existing').map((y) => y.existing);
+  initRoles() {
+    const { allRoles, user }                                   = this;
+    const { globalRoleBindings: usersGlobalRoleBindings = [] } = user;
+    const visibleRoles                                         = allRoles.filterBy('isHidden', false);
+    const baseRoles                                            = [ADMIN, USER, BASE].map((spr) => allRoles.findBy('id', spr))
```

Comment:
What's spr?

Reply:
renamed the variable `baseRoles` from `specialRoles` and forgot to change `spr` to `r`